tpe/plugin/src/SDFFeatures.cc

```diff
@@ -168,5 +168,27 @@ Identity SDFFeatures::ConstructSdfCollision(
   // and passed in as argument as there is no logic for searching resources
   // in ign-physics
   const auto collisionIdentity = this->AddCollision(link->GetId(), *collision);
+
+  // set collide bitmask
+  uint16_t collideBitmask = 0xFF;
+  if (_sdfCollision.Element())
+  {
+    // TODO(anyone) add category_bitmask as well
+    auto elem = _sdfCollision.Element();
+    if (elem->HasElement("surface"))
+    {
+      elem = elem->GetElement("surface");
+      if (elem->HasElement("contact"))
+      {
+        elem = elem->GetElement("contact");
+        if (elem->HasElement("collide_bitmask"))
+        {
+          collideBitmask = elem->Get<unsigned int>("collide_bitmask");
+          this->SetCollisionFilterMask(collisionIdentity, collideBitmask);
+        }
+      }
+    }
+  }
```

Comment:
Just a note on this (you are probably aware of it already), the original implementation in the dart SDFFeatures was done before the availability of optional features and the ign-gazebo PR that takes care of this logic using the [SDF Surface DOM](https://github.com/ignitionrobotics/ign-gazebo/pull/160/files). Admittedly when I did that PR I didn't double check that it could be compatible with Citadel, if that's the case we could probably backport it and get rid of this logic?

Reply:
I backported the collide bitmask changes to citadel in ignitionrobotics/ign-gazebo#221

I just noticed that if the logic here is removed, ign-physics alone won't be able to consume the collide bitmasks flags from sdf as it will now rely on ign-gazebo to set these flags, This causes ` SimulationFeature_TEST.CollideBitmasks` to fail unless we write extra logic in the integration test to set them. What do you think about leaving the logic here for ign-physics users that do not use ign-gazebo? I think it'll just be overriden by ign-gazebo as it parses SDF surface DOMs?

Reply:
Admittedly I don't know much about use cases for ignition libraries (such as physics, or rendering) without ignition gazebo, but no strong feelings here we can leave it if it helps with testing.

Reply:
ok yeah let's leave it in there for testing purposes.